components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/api/model/Webhook.java

```diff
@@ -0,0 +1,347 @@
+import org.wso2.carbon.identity.webhook.management.internal.service.impl.WebhookManagementServiceImpl;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Model class for Webhook subscription with two builders.
+ */
+public class Webhook {
+
+    private final String uuid;
+    private final String endpoint;
+    private final String description;
+    private final String secret;
+    private final int tenantId;
+    private final String eventSchemaName;
+    private final String eventSchemaUri;
+    private final WebhookStatus status;
+    private final Timestamp createdAt;
+    private final Timestamp updatedAt;
+    private final List<String> eventsSubscribed;
+
+    private Webhook(Builder builder) {
+
+        this.uuid = builder.uuid;
+        this.endpoint = builder.endpoint;
+        this.description = builder.description;
+        this.secret = builder.secret;
+        this.tenantId = builder.tenantId;
+        this.eventSchemaName = builder.eventSchemaName;
+        this.eventSchemaUri = builder.eventSchemaUri;
+        this.status = builder.status;
+        this.createdAt = builder.createdAt;
+        this.updatedAt = builder.updatedAt;
+        this.eventsSubscribed = builder.eventsSubscribed;
+    }
+
+    private Webhook(BuilderWithoutSecret builder) {
+
+        this.uuid = builder.uuid;
+        this.endpoint = builder.endpoint;
+        this.description = builder.description;
+        this.secret = null;
+        this.tenantId = builder.tenantId;
+        this.eventSchemaName = builder.eventSchemaName;
+        this.eventSchemaUri = builder.eventSchemaUri;
+        this.status = builder.status;
+        this.createdAt = builder.createdAt;
+        this.updatedAt = builder.updatedAt;
+        this.eventsSubscribed = builder.eventsSubscribed;
+    }
+
+    public String getUuid() {
+
+        return uuid;
+    }
+
+    public String getEndpoint() {
+
+        return endpoint;
+    }
+
+    public String getDescription() {
+
+        return description;
+    }
+
+    public String getSecret() {
+
+        return secret;
+    }
+
+    public int getTenantId() {
+
+        return tenantId;
+    }
+
+    public String getEventSchemaName() {
+
+        return eventSchemaName;
+    }
+
+    public String getEventSchemaUri() {
+
+        return eventSchemaUri;
+    }
+
+    public WebhookStatus getStatus() {
+
+        return status;
+    }
+
+    public Timestamp getCreatedAt() {
+
+        return createdAt;
+    }
+
+    public Timestamp getUpdatedAt() {
+
+        return updatedAt;
+    }
+
+    public List<String> getEventsSubscribed() throws WebhookMgtException {
+
+        if (eventsSubscribed != null) {
+            return eventsSubscribed;
+        }
+        return WebhookManagementServiceImpl.getInstance()
+                .getWebhookEvents(getUuid(), IdentityTenantUtil.getTenantDomain(tenantId));
```

Comment:
Let's populate the event list in the model from the list from service. so, if this function gets called multiple times from the model, it will cache the events rather than pulling from the service again

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6741/commits/153ddd354bdaf741877b9616f430bbae3d4b71b5